go/libkb/secret_store_device_ek.go

```diff
@@ -0,0 +1,123 @@
+package libkb
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// TODO consolidate these types when deviceEK generation is merged
+const EphemeralSeedLen = 32
+
+type DeviceEphemeralSeed [EphemeralSeedLen]byte
+type DeviceEKGeneration uint
+
+type SecretStoreDeviceEK struct {
+	username NormalizedUsername
+	ss       *SecretStoreLocked
+}
+
+func NewSecretStoreDeviceEK(g *GlobalContext) *SecretStoreDeviceEK {
+	return &SecretStoreDeviceEK{
+		ss:       NewSecretStoreLocked(g),
+		username: g.Env.GetUsername(),
+	}
+}
+
+func newDeviceEphermeralSeedFromBytes(b []byte) (deviceSecret DeviceEphemeralSeed, err error) {
+	if len(b) != EphemeralSeedLen {
+		err = fmt.Errorf("Wrong EphemeralSeedLen len: %d != %d", len(b), EphemeralSeedLen)
+		return deviceSecret, err
+	}
+	copy(deviceSecret[:], b)
+	return deviceSecret, nil
+}
+
+func deviceEmphemeralSeedFromLKSecFullSecret(f LKSecFullSecret) (deviceSecret DeviceEphemeralSeed) {
+	if f.f == nil {
+		return deviceSecret
+	}
+	return DeviceEphemeralSeed(*f.f)
+}
+
+func deviceEmphemeralSeedToLKSecFullSecret(seed DeviceEphemeralSeed) (secret LKSecFullSecret, err error) {
+	return newLKSecFullSecretFromBytes(seed[:])
+}
+
+func (s *SecretStoreDeviceEK) deviceEKPrefix() string {
+	return fmt.Sprintf("%s-device-ephemeral-key-", s.username)
+}
+
+func (s *SecretStoreDeviceEK) deviceEKName(generation DeviceEKGeneration) NormalizedUsername {
+	return NewNormalizedUsername(fmt.Sprintf("%s%d", s.deviceEKPrefix(), generation))
+}
+
+func (s *SecretStoreDeviceEK) RetrieveSecret(generation DeviceEKGeneration) (deviceSecret DeviceEphemeralSeed, err error) {
+	name := s.deviceEKName(generation)
+	secret, err := s.ss.RetrieveSecret(name)
+	if err != nil {
+		return deviceSecret, err
+	}
+	return deviceEmphemeralSeedFromLKSecFullSecret(secret), nil
+}
+
+func (s *SecretStoreDeviceEK) StoreSecret(generation DeviceEKGeneration, deviceSecret DeviceEphemeralSeed) (err error) {
+	name := s.deviceEKName(generation)
+	secret, err := deviceEmphemeralSeedToLKSecFullSecret(deviceSecret)
+	if err != nil {
+		return err
+	}
+	return s.ss.StoreSecret(name, secret)
+}
+
+func (s *SecretStoreDeviceEK) ClearSecret(generation DeviceEKGeneration) error {
+	name := s.deviceEKName(generation)
+	return s.ss.ClearSecret(name)
+}
+
+func (s *SecretStoreDeviceEK) getGenerations() (generations []DeviceEKGeneration, err error) {
+	usernames, err := s.ss.GetUsersWithStoredSecrets()
```

Comment:
This seems a little weird to me since we're overloading the device EKs as usernames. Should we make a `GetDevicesWithStoredSecrets` method part of the `SecretStore` interface and filter deviceEKs from the other implementations? @patrickxb what is your opinion?